### DISCUSSION TITLE
SyncFolderItems operation base only on calendar_id

### DISCUSSION THREAD

**Discussion by suhanoves** [ecederstrand/exchangelib#1054] [Q&A] [answered]
Hello, Erik, I need your help again!

TLDR: How to call the `.sync_items` method without connecting to Exchange to get the real calendar instance? Is it possible to create a local `fake_calendar` only with `id` and `account` attributes for this?

I have a `calendar_id` string of a subcalendar and I would like to `.sync_items` this subcalendar.
I know that I can get an instance by id using `SingleFolderQuerySet` and call the method - but it will create a query to Exchange for getting calendar instance.

I thought I could create a `fake_calendar = Folder(id=my_id, account=my_account)` because id is enough the SyncFolderItems operation to work, but `AttributeError(f"{sorted(kwargs.keys())!r} are invalid kwargs for this class")` was raised.
Another way was `calendar = Folder(parent=account.calendar, id='my_id')` but it also create a query to Exchange to get calendar instance.

Is it even possible by chance to use the sync_items method only base on calendar_id (which I received from the subscription Notification) and immediately get a response in the form of a generator of occurred event changes?

**Comment by ecederstrand** [comment]
`Folder(id=my_id, account=my_account)` would be the way to go. The `AttributeError` hints that you're passing invalid keyword arguments somewhere, but you'll have tp post the entire code and stack trace to see where.

**Reply by suhanoves** [reply]
@ecederstrand thanks for the quick response and sorry for my silence, was on vacation.

For some reason I can't create a Folder this way. I got an Account and even checked that it exists. Next line, I try to create a Folder and immediately get an error `AttributeError: ['account'] are invalid kwargs for this class`. The initializer of `class EWSElement(metaclass=EWSMeta)` does not allow me to set an `account` attribute.

```python
my_account = Account(
    primary_smtp_address="my_user@my_exchange.com",
    config=config,
    access_type=IMPERSONATION,
    autodiscover=False,
)  # connection correctly
root_cal = my_account.calendar  # successfully receive the Calendar
cal = Folder(id=some_real_id, account=my_account)  # got an Exception
```
Traceback:
```
  File "...", line 37, in <module>
    cal = Folder(id=some_real_id, account=my_account)
  File ".../exchangelib/folders/base.py", line 793, in __init__
    super().__init__(**kwargs)
  File ".../exchangelib/folders/base.py", line 90, in __init__
    super().__init__(**kwargs)
  File ".../exchangelib/properties.py", line 1453, in __init__
    super().__init__(**kwargs)
  File ".../properties.py", line 263, in __init__
    raise AttributeError(f"{sorted(kwargs.keys())!r} are invalid kwargs for this class")
AttributeError: ['account'] are invalid kwargs for this class
```
Are there any other ways?

**Reply by ecederstrand** [reply]
Sorry, the way to create a `Folder` instance is either `Folder(id=some_real_id, root=account.root)` or `Folder(id=some_real_id, parent=my_account.calendar)`

**Comment by suhanoves** [comment, marked as answer]
For some reason, the subcomment cannot be marked as an answer, so I'll copy it here

> Sorry, the way to create a `Folder` instance is either `Folder(id=some_real_id, root=account.root)` or `Folder(id=some_real_id, parent=my_account.calendar)`
